source/sessions/tests/snapshot-sessions.yml

```diff
@@ -480,3 +481,404 @@ tests:
       isError: true
       isClientError: true
       errorContains: Transactions are not supported in snapshot sessions
+
+- description: Find operation with snapshot and snapshot time
+  operations:
+    - name: find
+      object: collection0
+      arguments:
+        session: session0
+        filter: {}
+      expectResult:
+        - { _id: 1, x: 11 }
+        - { _id: 2, x: 11 }
+    - name: getSnapshotTime
+      object: session0
+      saveResultAsEntity: &savedSnapshotTime savedSnapshotTime
+    - name: insertOne
+      object: collection0
+      arguments:
+        document: { _id: 3, x: 33 }
+    - name: createEntities
+      object: testRunner
+      arguments:
+        entities:
+          - session:
+              id: session2
+              client: client0
+              sessionOptions:
+                snapshot: true
+                snapshotTime: *savedSnapshotTime
+    - name: find
+      object: collection0
+      arguments:
+        session: session2
+        filter: {}
+      expectResult:
+        - { _id: 1, x: 11 }
+        - { _id: 2, x: 11 }
+    - name: find
```

Comment:
Why do we making `find` with the `session2` twice, or have I missed some differences in the operations? If this is because we want to make sure `snapshot` time will not be changes after the first call - then it might makes sense to add comments here (same with other tests below).